.gitignore

```diff
@@ -1,6 +1,54 @@
-target/
+/target/
+.gradle
+build/
+!**/src/main/**/target/
+!**/src/test/**/target/
+!**/src/main/**/build/
+!**/src/test/**/build/
 
-# The following files are generated/updated by Hilla
+.DS_Store
+
+# The following files are automatically generated/updated
 node_modules/
 frontend/generated/
+pnpmfile.js
+.npmrc
+webpack.generated.js
 vite.generated.ts
+
+# Eclipse and STS
+.apt_generated
+.classpath
+.factorypath
+.project
+.settings
+.springBeans
+.sts4-cache
+
+# IntelliJ IDEA
+.idea
+*.iws
+*.iml
+*.ipr
+out/
+!**/src/main/**/out/
+!**/src/test/**/out/
+
+# NetBeans
+/nbproject/private/
+/nbbuild/
+/dist/
+/nbdist/
+/.nb-gradle/
+
+# VS Code
+.vscode/
+
+# Maven wrapper
+.mvn/
+mvnw
+
+# The following are auto-generated if missing, but they should be added to source control if customized.
+tsconfig.json
+types.d.ts
+index.html
```

Comment:
Looks like nonsense to me, keeping these sources under control is absolutely necessary in Hilla apps, regardless of the auto-regeneration features.

```suggestion
```